### PR TITLE
add gray image support

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?
add gray image support

## How was this patch tested?
unit test

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/2219

### CODE REVIEW

**Comment by SeaOfOcean** [comment]
pr validation pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/Convertor.scala [hunk 1]
```diff
@@ -69,8 +69,12 @@ object BytesToMat {
 class PixelBytesToMat(byteKey: String = ImageFeature.bytes) extends FeatureTransformer {
 
   override def transformMat(feature: ImageFeature): Unit = {
+    require(feature.getOriginalSize != null,
+      "please set the original size of image in ImageFeature")
     val pixels = feature[Array[Byte]](byteKey)
-    val mat = OpenCVMat.fromPixelsBytes(pixels, feature.getOriginalHeight, feature.getOriginalWidth)
+    val mat = OpenCVMat.fromPixelsBytes(pixels, feature.getOriginalHeight,
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/Convertor.scala]: why not using feature.getOriginalSize._1 feature.getOriginalSize._2 to keep consistent with the way channel is fetched

Comment (wzhongyuan) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/Convertor.scala]: Or maybe add a new method called getChannel ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/ImageFeature.scala [hunk 2]
```diff
@@ -200,12 +200,22 @@ class ImageFeature extends Serializable {
    */
   def copyTo[T: ClassTag](storage: Array[T], offset: Int, floatKey: String = ImageFeature.floats,
     toRGB: Boolean = true)(implicit ev: TensorNumeric[T]): Unit = {
+    val channel = getChannel()
     require(contains(floatKey), s"there should be ${floatKey} in ImageFeature")
     val data = floats(floatKey)
-    require(data.length >= getWidth() * getHeight() * 3,
+    require(data.length >= getWidth() * getHeight() * channel,
       s"float array length should be larger than 3 * ${getWidth()} * ${getHeight()}")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/ImageFeature.scala]: float array length should be larger than 3 * ${getWidth()} * ${getHeight()} => float array length should be larger than  ${getChannel() ? }* ${getWidth()} * ${getHeight()} *

**Comment by wzhongyuan** [comment]
LGTM

**Pull request merged**